lib/teleterm/daemon/daemon.go

```diff
@@ -575,6 +575,15 @@ func (s *Service) Stop() {
 		gateway.Close()
 	}
 
+	s.usageReporter.GracefulStop()
+
+	select {
+	case <-s.usageReporter.Closer.C:
+		s.cfg.Log.Info("Gracefully stopped usage reporter.")
+	case <-time.After(time.Second * 5):
+		s.cfg.Log.Info("Gracefully stopping usage reporter timed out.")
+	}
```

Comment:
I mentioned that in the DMs, but instead of exposing `Closer` and the caller having to orchestrate all of this, it might be better to make `GracefulStop` similar to [`net.http.Server.Shutdown`](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=2772;drc=1c65b69bd1dbc930c6246877f6c21c81f2a60d55). `GracefulStop` could take a context and then select on either `r.gracefulStopScheduled <- true` or the context being closed.

Then the caller could pass a context with a timeout and handle the error returned from `GracefulStop` in case the usage reported didn't stop gracefully due to a timeout of the context.